sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -364,21 +364,28 @@ private static IReadOnlyList<ServiceBusReceivedMessage> EmptyList
                         .ConfigureAwait(false);
                 }
 
-                var messagesReceived = await link.ReceiveMessagesAsync(
+                // Rely on the fact that the underlying IEnumerable is actually a collection. If this changes, our
+                // tests will fail.
+                var messagesReceived = (IReadOnlyCollection<AmqpMessage>) await link.ReceiveMessagesAsync(
```

Comment:
I'm torn on whether to just nod because it's unlikely to change or suggest we do a switch to transform if it's not a collection to protect ourselves because it's not the part of the contract.

Since this is in the AMQP library and that may be hoisted for applications using it directly or with other packages, we probably should add a guard transform to avoid "but it worked yesterday" scenarios, however unlikely.